### PR TITLE
#155319173, #155319231, #155183816, #155922706 Add the navigation view - OnHold

### PR DESCRIPTION
**Pull request by muhallan**
#### What does this PR do?
Adds the navigation view to the SerialEntryActivity
#### Description of Task to be completed?
- Ensure the hamburger icon is visible
- Add clickable menu items to the navigation view
- Write tests for the navigation view
#### How should this be manually tested?
Checkout to this branch, run the app, then click hamburger icon, or swipe right to draw the navigation menu and swipe left to close it.
#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/155319173
- https://www.pivotaltracker.com/story/show/155319231
- https://www.pivotaltracker.com/story/show/155183816
#### Screenshots (if appropriate)
Hamburger Icon
image

Navigation Menu
image

### CODE REVIEW

File: app/src/main/res/layout/nav_header_navigation_drawer.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    android:paddingBottom="@dimen/activity_vertical_margin"
+    android:paddingLeft="@dimen/activity_horizontal_margin"
+    android:paddingRight="@dimen/activity_horizontal_margin"
+    android:paddingTop="@dimen/activity_vertical_margin"
+    android:theme="@style/ThemeOverlay.AppCompat.Dark">
+
+
+    <de.hdodenhof.circleimageview.CircleImageView
+        android:id="@+id/displayPicture"
+        android:layout_width="80dp"
```
Comment (suada-haji) [inline app/src/main/res/layout/nav_header_navigation_drawer.xml]: Place values in the `dimens` resource file

File: app/src/main/res/layout/app_bar_navigation_drawer.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    <android.support.design.widget.AppBarLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:fitsSystemWindows="true"
+        app:elevation="0dp"
+        android:theme="@style/AppTheme.AppBarOverlay">
+
+        <android.support.v7.widget.Toolbar
+            android:id="@+id/toolbarTB"
```
Comment (suada-haji) [inline app/src/main/res/layout/app_bar_navigation_drawer.xml]: Refrain from using `Caps` in ids

File: app/src/main/res/layout/nav_header_navigation_drawer.xml (hunk 1 above)
Comment (Ewanjiru) [reply app/src/main/res/layout/nav_header_navigation_drawer.xml]: well noted.

File: app/src/main/res/layout/app_bar_navigation_drawer.xml (hunk 2 above)
Comment (Ewanjiru) [reply app/src/main/res/layout/app_bar_navigation_drawer.xml]: Well noted.

File: app/src/main/java/com/andela/art/serialentry/SerialEntryActivity.java [hunk 3]
```diff
@@ -21,7 +21,7 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_serial_entry);
 
-        Toolbar toolbar = findViewById(R.id.toolbarTB);
+        Toolbar toolbar = findViewById(R.id.toolbar);
```
Comment (suada-haji) [inline app/src/main/java/com/andela/art/serialentry/SerialEntryActivity.java]: You could implement Data binding here, you can reach to @asce1062  to show you how they did it

Comment (Ewanjiru) [reply app/src/main/java/com/andela/art/serialentry/SerialEntryActivity.java]: well noted.

**Pull request closed without merging**